### PR TITLE
Small fixes

### PR DESCRIPTION
**Pull request by Bob-AT**
* fix some typos
* fix some constructor bugs
* minor refactoring of Objectives that I need for _Kill Confirmed (Semi-Perm.)_

### CODE REVIEW

**Review by JakBaranowski — COMMENTED** [review]
@Bob-AT thank you for submitting this PR, there's a bunch of improvements here. However, I've added a few comments that need addressing. 

That being said I understand that this PR is 8 months old, your fork of the project already moved on, and I would understand if you're not willing to make any additional changes to this PR. If that's the case, I'll gladly implement the changes myself and merge the PR.

Also, sorry for not looking into this PR earlier.

File: GroundBranch/Content/GroundBranch/Lua/Objectives/ConfirmKill.lua [hunk 1]
```diff
@@ -89,6 +89,15 @@ function ConfirmKill:Reset()
 	self.HVT.EliminatedAndConfirmedCount = 0
 end
 
+function ConfirmKill:GetCompletedObjectives()
```
[the remark this answers is not in the document]
Comment (Bob-AT) [reply GroundBranch/Content/GroundBranch/Lua/Objectives/ConfirmKill.lua:92]: Yes, good idea. I would have the objective name be defined in `ConfirmKill`

File: GroundBranch/Content/GroundBranch/Lua/Objectives/ConfirmKill.lua [hunk 2]
```diff
@@ -49,9 +51,7 @@ function ConfirmKill:Create(
     hvtTag,
     hvtCount
 )
-    local killConfirmation = {}
-    setmetatable(killConfirmation, self)
-    self.__index = self
+    local self = setmetatable({}, ConfirmKill)
```
[the remark this answers is not in the document]
Comment (Bob-AT) [reply GroundBranch/Content/GroundBranch/Lua/Objectives/ConfirmKill.lua:54]: No objection to `local object = ...` or `local instance`

File: GroundBranch/Content/GroundBranch/Lua/Players/Teams.lua [hunk 3]
```diff
@@ -13,22 +13,23 @@ local Teams = {
         ScoreMessage = false,
         ScoreMilestone = true,
         ObjectiveMessage = true,
-        ObjectivePrompt = true
+        ObjectivePrompt = true,
+        Always = true
```
[the remark this answers is not in the document]
Comment (Bob-AT) [reply GroundBranch/Content/GroundBranch/Lua/Players/Teams.lua:17]: `Always` is need for my Kill Confirmed Semi-Permissve

**Comment by Bob-AT** [comment]
This PR is slightly out of date.
I've addressed the technical concerns.

Perhaps it would make sense to VC on Discord to see how we can collaborate in the future.
From my perspective it does not make sense to have two forks as we would just waste engineering time.

**Pull request closed without merging**